storage/innobase/handler/ha_innodb.cc

```diff
@@ -10977,18 +10977,44 @@ create_table_info_t::create_table_def()
 			dict_mem_table_free(table);
 		} else if (m_create_info->encrypt_type.length > 0
 			   && !Encryption::is_none(
-				   m_create_info->encrypt_type.str)) {
+				   m_create_info->encrypt_type.str)
+			   && !srv_tmp_tablespace_encrypt) {
 
 			my_error(ER_TABLESPACE_CANNOT_ENCRYPT, MYF(0));
 			err = DB_UNSUPPORTED;
 			dict_mem_table_free(table);
 		} else {
+			if (srv_tmp_tablespace_encrypt) {
+				/* force encryption for temporary tables */
+				byte*			master_key = NULL;
+				ulint			master_key_id;
+				Encryption::Version	version;
+
+				/* Check if keyring is ready. */
+				Encryption::get_master_key(&master_key_id,
+							   &master_key,
+							   &version);
+
+				if (master_key == NULL) {
+					my_error(ER_CANNOT_FIND_KEY_IN_KEYRING,
+						 MYF(0));
+					err = DB_UNSUPPORTED;
+					dict_mem_table_free(table);
+				} else {
+					my_free(master_key);
+					DICT_TF2_FLAG_SET(table,
+							  DICT_TF2_ENCRYPTION);
+				}
+			}
+
+			if (err == DB_SUCCESS) {
 
-			/* Get a new table ID */
-			dict_table_assign_new_id(table, m_trx);
+				/* Get a new table ID */
+				dict_table_assign_new_id(table, m_trx);
```

Comment:
I wonder why github shows the green bars here but not below. Tabs vs spaces?

Reply:
No, just one more level of indentation added